src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -53,97 +62,124 @@ export class EventStoreComponent implements OnInit, OnDestroy {
           this.isResetEnabled = true;
-          this.eventsList.reload().subscribe( data => {
-              this.setTimelineData();
-          });
-      } else {
-          this.resetSelectionProperties();
+          return data.eventsList.reload();
+      });
+
+      if (subs) {
+          forkJoin(subs).subscribe(() => this.setTimelineData());
       }
   }
 
+  public mergeTimelineData(combinedData: ITimelineData, data: ITimelineData): void {
+      data.items.forEach(item => combinedData.items.add(item));
+
+      data.groups.forEach(group => combinedData.groups.add(group));
+
+      combinedData.potentiallyMissingEvents =
+          combinedData.potentiallyMissingEvents || data.potentiallyMissingEvents;
+  }
+
+  private initializeTimelineData(): ITimelineData {
+      return {
+          start: this.startDate,
+          end: this.endDate,
+          groups: new DataSet<DataGroup>(),
+          items: new DataSet<DataItem>()
+      };
+  }
+
   public setTimelineData(): void {
-    this.eventsList.ensureInitialized().subscribe( () => {
-        try {
-            if (this.pshowAllEvents) {
-                const d = parseEventsGenerically(this.eventsList.collection.map(event => event.raw), this.transformText);
-
-                this.timeLineEventsData = {
-                    groups: d.groups,
-                    items: d.items,
-                    start: this.startDate,
-                    end: this.endDate,
-                    potentiallyMissingEvents: d.potentiallyMissingEvents
-                };
-
-            }else if (this.timelineGenerator) {
-                const d = this.timelineGenerator.generateTimeLineData(this.eventsList.collection.map(event => event.raw), this.startDate, this.endDate);
-
-                this.timeLineEventsData = {
-                    groups: d.groups,
-                    items: d.items,
-                    start: this.startDate,
-                    end: this.endDate,
-                    potentiallyMissingEvents: d.potentiallyMissingEvents
-                };
-            }
-        }catch (e) {
-            console.error(e);
-        }
-    });
-    }
-
-    setNewDates(dates: IOnDateChange) {
-        this.debounceHandler.next(dates);
-    }
+      let rawEventlist = [];
+      const combinedTimelineData = this.initializeTimelineData();
+
+      const subs = this.listEventStoreData.map(data => data.eventsList.ensureInitialized());
+
+      forkJoin(subs).subscribe(() => {
+          for (const data of this.listEventStoreData) {
+              try {
+                  if (this.pshowAllEvents) {
+                      rawEventlist = rawEventlist.concat(data.eventsList.collection.map(event => event.raw));
+
+                  } else if (data.timelineGenerator) {
+                      const d = data.timelineGenerator.generateTimeLineData(data.eventsList.collection.map(event => event.raw), this.startDate, this.endDate);
+
+                      data.timelineData = {
```

Comment:
so this should be directly assignable, looking at it now there was no reason to pick the properties off before. const d doesnt seem necessary and could be directly assigned to the timeline data